server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessAttributesHealth.java

```diff
@@ -0,0 +1,289 @@
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContentFragment;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * Cluster Awareness health information
+ *
+ */
+public class ClusterAwarenessAttributesHealth implements Iterable<ClusterAwarenessAttributeValueHealth>, Writeable, ToXContentFragment {
+
+    private final String awarenessAttributeName;
+    private Map<String, ClusterAwarenessAttributeValueHealth> awarenessAttributeValueHealthMap;
+
+    /**
+     * Creates Awareness AttributeValue Health information
+     *
+     * @param awarenessAttributeValue Awareness Attribute value ie zone, rack etc
+     * @param displayUnassignedShardLevelInfo Governs if unassigned info should be visible or not
+     * @param clusterState cluster state
+     */
+    public ClusterAwarenessAttributesHealth(
+        String awarenessAttributeValue,
+        boolean displayUnassignedShardLevelInfo,
+        ClusterState clusterState
+    ) {
+        awarenessAttributeName = awarenessAttributeValue;
+
+        // This is the Map which is storing the per attribute value health stats.
+        // The key would be the attribute value like rack-1 and the stats would be stored as value.
+        Map<String, ClusterAwarenessAttributeValueHealth> attributesHealth = new HashMap<>();
+
+        // Getting the node map for cluster
+        ImmutableOpenMap<String, DiscoveryNode> nodeMap = clusterState.nodes().getDataNodes();
+
+        // This is the map that would store all the stats per attribute ie
+        // health stats for rack-1, rack-2 etc.
+        awarenessAttributeValueHealthMap = new HashMap<>(Collections.emptyMap());
+        String attributeValue;
+
+        if (!nodeMap.isEmpty()) {
+            Iterator<String> iter = nodeMap.keysIt();
+            while (iter.hasNext()) {
+                ClusterAwarenessAttributeValueHealth clusterAwarenessAttributeValueHealth = null;
+                String node = iter.next();
+                DiscoveryNode nodeDiscovery = nodeMap.get(node);
+                Map<String, String> nodeAttributes = nodeDiscovery.getAttributes();
+                if (!nodeAttributes.isEmpty()) {
+                    if (nodeAttributes.containsKey(awarenessAttributeName)) {
+                        attributeValue = nodeAttributes.get(awarenessAttributeName);
+
+                        if (!attributesHealth.containsKey(attributeValue)) {
+                            clusterAwarenessAttributeValueHealth = new ClusterAwarenessAttributeValueHealth(attributeValue);
```

Comment:
Lets pass the RoutingNode here to avoid unnecessary computation

Reply:
computation will still exists in the parent class . However will make the required change